### PR TITLE
Detect requirements.txt and present the user with the option to force generation of it + better handling of no rsconnect found case + rsconnect-python version bump

### PR DESCRIPTION
**Pull request by scottmmjackson**
### Description

- Requires https://github.com/rstudio/rsconnect-python/pull/71
- When `rsconnect-python` not found, present an appropriate error:

Screen Shot 2020-03-03 at 9 22 52 AM

- Tie `rsconnect-jupyter` to at least version 1.3.2 of `rsconnect-python` to enable the use of the `force-generate` option in environment (and, subsequently, `compatibility-mode`, to be addressed in a follow-on)

- Implement interfaces in the `Write Manifest` and `Publish` dialogs that permit `requirements.txt` to be overwritten. Note that in the future we may want do more when we prefetch their existence and react a lot smarter, but this is requisite work to support conda environment export from jupyter.

- Huge update of the python side to match the reworked API surface in `rsconnect-python`

- Readded test pypi install to the test script. We need this unless we want to only ever develop against a released version of `rsconnect-python`

### Testing Notes / Validation Steps

This PR requires a complete revalidation of pretty much every workflow in the plugin:

- Add a server
- Add a server with custom certificate
- Add a server which can only be added by disabling TLS
- Deploy new content
- Deploy existing content
- Deploy content with files added

Finally, verifying the actual feature change:
- In deploy dialog and write-manifest dialog
  - If `requirements.txt` isn't present, a message to that effect should be presented
  - If `requirements.txt` is present, radio buttons should be presented asking the user to choose
  - In the radio button case, both radios should work as expected.

Note that if this feature is validated after conda support has landed, that PR will have a somewhat modified version of the testing workflow immediately previous.

Known Issues: I suspect that something changed with the `app_search` surface that's making us return more results than previously. I haven't completely figured it out yet but I'm treating it as something to fix after this PR.

### CODE REVIEW

File: rsconnect_jupyter/__init__.py [hunk 1]
```diff
@@ -76,36 +78,49 @@ def post(self, action):
...
 
             try:
-                retval = app_search(uri, api_key, title, app_id, disable_tls_check, cadata)
+                with RSConnect(
+                    RSConnectServer(
+                        uri,
+                        api_key,
+                        disable_tls_check,
+                        cadata
+                    )
```
Comment (mmarchetti) [inline rsconnect_jupyter/__init__.py]: the formatting here might benefit from a temporary variable

File: rsconnect_jupyter/static/connect.js [hunk 2]
```diff
@@ -1626,14 +1713,48 @@ define([
...
+        ContentsManager.list_contents(notebookDirectory)
+            .then(function (contents) {
+              for(var index in contents.content) {
+                if (contents.content[index].name === 'requirements.txt') {
+                  hasRequirementsTxt = true;
+                  break;
+                }
+              }
+              prepareRequirementsTxtDialog(hasRequirementsTxt);
+            });
```
Comment (mmarchetti) [inline rsconnect_jupyter/static/connect.js]: This seems to duplicate the code on lines 1112-1122 - should this be factored out?

File: rsconnect_jupyter/static/rsconnect.js [hunk 3]
```diff
@@ -265,13 +265,18 @@ define([
...
                 try {
+                    var flags = '';
+                    if (compatibilityMode || forceGenerate) {
+                        flags = '-' + (compatibilityMode ? 'c' : '') + (forceGenerate ? 'f' : '');
+                    }
                     var cmd = [
                         '!"',
                         pythonPath,
-                        '" -m rsconnect.environment "${PWD}"'
+                        '" -m rsconnect.environmentt ' +
```
Comment (mmarchetti) [inline rsconnect_jupyter/static/rsconnect.js]: looks like a typo here

File: rsconnect_jupyter/static/connect.js (hunk 2 above)
Comment (scottmmjackson) [reply rsconnect_jupyter/static/connect.js]: It's different, I'll rename them to make that clearer

**Pull request merged**